modules/caddyhttp/reverseproxy/streaming.go

```diff
@@ -137,9 +137,9 @@ func (h Handler) copyResponse(dst io.Writer, src io.Reader, flushInterval time.D
 		}
 	}
 
-	buf := streamingBufPool.Get().([]byte)
+	buf := streamingBufPool.Get().(*[]byte)
 	defer streamingBufPool.Put(buf)
-	_, err := h.copyBuffer(dst, src, buf)
+	_, err := h.copyBuffer(dst, src, *buf)
```

Comment:
`*[]byte` is certainly awkward vs `[]byte`, but makes for almost a 50% reduction in allocations (see comment below), and speeds this up too.

I benchmarked this and got these results:

```
BenchmarkHandlerCopyResponseOld-8   	   31369	     37921 ns/op	  132364 B/op	       8 allocs/op
BenchmarkHandlerCopyResponse-8      	   31453	     36648 ns/op	  132209 B/op	       5 allocs/op
```

Note that I also tried out a `*bytes.Buffer` (which is the usual approach, and faster/fewer allocs than a `[]byte`) but using a `*[]byte` is fastest.

Reply:
Oh yes, SA6002 indeed. I didn't commit the benchmarks since they would be referencing functions that I deleted, but here they are:

```go
func BenchmarkHandlerCopyResponseOld(b *testing.B) {
	h := Handler{}

	testdata := []string{
		"",
		strings.Repeat("a", defaultBufferSize),
		strings.Repeat("123456789 123456789 123456789 12", 3000),
	}
	dst := new(bytes.Buffer)
	for i := 0; i < b.N; i++ {
		for _, d := range testdata {
			src := bytes.NewBuffer([]byte(d))
			dst.Reset()
			h.copyResponseOld(dst, src, 0)
		}
	}
}

func BenchmarkHandlerCopyResponse(b *testing.B) {
	h := Handler{}
	testdata := []string{
		"",
		strings.Repeat("a", defaultBufferSize),
		strings.Repeat("123456789 123456789 123456789 12", 3000),
	}
	dst := new(bytes.Buffer)
	for i := 0; i < b.N; i++ {
		for _, d := range testdata {
			src := bytes.NewBuffer([]byte(d))
			dst.Reset()
			h.copyResponse(dst, src, 0)
		}
	}
}
```

I agree about the insignificance of the speedup - but the allocation reduction was the main reason I included the change 😉